addons/l10n_ec/migrations/3.4/post-migration.py

```diff
@@ -0,0 +1,87 @@
+    ''')
+    env.cr.execute('''
+        update account_tax
+        set tax_group_id=t.id from (select id from account_tax_group where l10n_ec_type='withhold_vat_purchase') as t
+        where account_tax.id in (select id from account_tax where tax_group_id in (select id from account_tax_group where l10n_ec_type='withhold_vat') and type_tax_use='purchase')
+    ''')
+    env.cr.execute('''
+        update account_tax
+        set tax_group_id=t.id from (select id from account_tax_group where l10n_ec_type='withhold_income_sale') as t
+        where account_tax.id in (select id from account_tax where tax_group_id in (select id from account_tax_group where l10n_ec_type='withhold_income_tax') and type_tax_use='sale')
+    ''')
+    env.cr.execute('''
+        update account_tax
+        set tax_group_id=t.id from (select id from account_tax_group where l10n_ec_type='withhold_income_purchase') as t
+        where account_tax.id in (select id from account_tax where tax_group_id in (select id from account_tax_group where l10n_ec_type='withhold_income_tax') and type_tax_use='purchase')
+    ''')
+
+def update_type_tax_use(env):
+    # sets type_tax_use = none for withholding taxes
+    env.cr.execute('''
+        update account_tax
+        set type_tax_use = 'none'
+        where tax_group_id in (select id from account_tax_group where l10n_ec_type in ('withhold_income_purchase','withhold_vat_purchase','withhold_income_sale','withhold_vat_sale'))
+    ''')
+
+def unlink_old_withhold_group(env):
+    deprecated_withhold_vat_group = env.ref('l10n_ec.tax_group_withhold_vat', False)
+    deprecated_withhold_profit_group = env.ref('l10n_ec.tax_group_withhold_income', False)
+    deprecated_withhold_vat_group and deprecated_withhold_vat_group.unlink()
+    deprecated_withhold_profit_group and deprecated_withhold_profit_group.unlink()
+
+def update_vat_withhold_base_percent(env):
+    # For vat withhold taxes, replace factor_percent=12% with factor_percent=100%
+    all_companies = env['res.company'].search([])
+    ecuadorian_companies = all_companies.filtered(lambda r: r.country_code == 'EC')
+    ecuadorian_taxes = env['account.tax'].search([('company_id', 'in', ecuadorian_companies.ids)])
+    taxes_to_fix = ecuadorian_taxes.filtered(lambda x: x.tax_group_id.l10n_ec_type in ['withhold_vat_sale', 'withhold_vat_purchase'])
+    env.cr.execute('''
+        --for invoice_tax_id
+        update account_tax_repartition_line
+        set factor_percent=100 
+        where factor_percent=12 
+        and repartition_type='tax'
+        and invoice_tax_id in %s
+        ''', [tuple(taxes_to_fix.ids)])
+    env.cr.execute('''
+        --for refund_tax_id
+        update account_tax_repartition_line
+        set factor_percent=100 
+        where factor_percent=12 
+        and repartition_type='tax'
+        and refund_tax_id in %s
+        ''', [tuple(taxes_to_fix.ids)])
+
+def recompute_invoice_names(env):
```

Comment:
I would avoid this method.  It seems unnecessary and will lead to more confusion than anything else.